src/azure-cli/azure/cli/command_modules/synapse/operations/workspace.py

```diff
@@ -69,8 +98,30 @@ def update_workspace(cmd, client, resource_group_name, workspace_name, sql_admin
     else:
         tenant_ids_list = allowed_aad_tenant_ids
 
+    if repository_type:
```

Comment:
If customer only want to update `collaboration_branch`, they may not specify `--repository-type`. Same concern with create operation.

Reply:
After team discussion, we trend to reply backend sever to make validation and avoid to make complex validation. So that when serve make change, synapse cli won't take many effort to keep working.

Reply:
What I mean is if customer run `az synapse workspace update --collaboration-branch`, nothing will change because `repository_type` here is `None`. Is that expected?

Reply:
yes, user need to specify all these parameters when update. Will improve help doc to let user better know.

Reply:
In that case, I'll suggest to use a validator for `repository_type` where the warning message can be added:
```
c.argument('repository_type', arg_group=repository_arg_group, arg_type=get_enum_type(['AzureDevOpsGit', 'GitHub']), validator=validate_repository_type, help='The repository configuration type.')
```